### PR TITLE
User can register an account

### PR DESCRIPTION
**Pull request by Mljungho**
### THIS PR CONTAINS: 

- Adding registration JSON fixtures
- Adding test specification for registration.
- Adding User registration page.
- Installing & Setting up J-tochauth
- Adding feature to navigate to login page.
- Login page is empty. 

https://www.pivotaltracker.com/story/show/181151517
@giacoletti @DefyCab 

image

image

### CODE REVIEW

File: cypress/fixtures/registration_successfull_response.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (giacoletti) [inline cypress/fixtures/registration_successfull_response.json]: misspelling in the file name:
"successful" is one L

File: cypress/fixtures/registrationPasswordMissmatch.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (giacoletti) [inline cypress/fixtures/registrationPasswordMissmatch.json]: misspelling in the file name:
"mismatch" is one S

File: cypress/integration/visitorCanRegisterAnAccount.feature.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      cy.visit("/");
+      cy.get("[data-cy=sign-up-btn]").click();
+      cy.get("[data-cy=user-name-input]").type("john");
+      cy.get("[data-cy=email-input]").type("johnskoglung@test.com");
+      cy.get("[data-cy=password-input]").type("password");
+      cy.get("[data-cy=conf-password-input]").type("password");
+      cy.get("[data-cy=register-button]").click();
+    });
+
+    it("is expected to dispaly registration success message", () => {
```
Comment (giacoletti) [inline cypress/integration/visitorCanRegisterAnAccount.feature.js]: "to display"

File: src/App.jsx [hunk 4]
```diff
@@ -13,6 +15,8 @@ const App = () => {
         <Routes>
           <Route path="/" element={<RecipesMainView />} />
           <Route path="recipes/:id" element={<RecipeFullView />} />
+          <Route path="registration-form" element={<RegistrationForm />} />
```
Comment (giacoletti) [inline src/App.jsx]: "registration-form" as a path it doesn't sound nice...
I would use something like "signup", cleaner and more concise.

File: src/modules/authentication.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import auth from "./auth";
```
Comment (giacoletti) [inline src/modules/authentication.js]: you can call the file name in pascal case, same name as the object is exporting: "Authentication"

**Review by giacoletti — CHANGES_REQUESTED** [review]

File: cypress/fixtures/registrationPasswordMissmatch.json (hunk 2 above)
Comment (Mljungho) [reply cypress/fixtures/registrationPasswordMissmatch.json]: Fixed

File: cypress/integration/visitorCanRegisterAnAccount.feature.js (hunk 3 above)
Comment (Mljungho) [reply cypress/integration/visitorCanRegisterAnAccount.feature.js]: "is expected"

File: src/App.jsx (hunk 4 above)
Comment (Mljungho) [reply src/App.jsx]: Followed advice and changed.

File: src/modules/authentication.js (hunk 5 above)
Comment (Mljungho) [reply src/modules/authentication.js]: Ok

File: cypress/fixtures/registrationPasswordMissmatch.json (hunk 2 above)
Comment (giacoletti) [reply cypress/fixtures/registrationPasswordMissmatch.json]:
```
registrationPasswordMissmatch.json
```
this is what I was referring to

File: cypress/integration/visitorCanRegisterAnAccount.feature.js (hunk 3 above)
Comment (giacoletti) [reply cypress/integration/visitorCanRegisterAnAccount.feature.js]: `dispaly`
it is misspelled

**Review by giacoletti — CHANGES_REQUESTED** [review]

**Pull request merged**